### PR TITLE
BUG: fix bugs in io.votable related to numpy 2's representation of scalars

### PR DESCRIPTION
**Pull request by neutrinoceros**
### Description
This pull request is to address part of #16423 (specifically, `io.votable`).
~The first commit is common with #16426, #16427 and #16433 (allowing them to be merged in arbitrary order).~
The fix itself is in part taken from #15065

- [ ] By checking this box, the PR author has requested that maintainers do **NOT** use the "Squash and Merge" button. Maintainers should respect this when possible; however, the final decision is at the discretion of the maintainer that merges the PR.

### CODE REVIEW

**Comment by pllim** [comment]
pyvo and/or astroquery might want to review this. I will ping them.

File: astropy/io/votable/tests/test_vo.py [hunk 1]
```diff
@@ -178,20 +178,23 @@ def _test_regression(tmp_path, _python_based=False, binary_mode=1):
 
 
 @pytest.mark.xfail("legacy_float_repr")
+@pytest.mark.usefixtures("without_legacy_printoptions")
```
Comment (mhvk) [inline astropy/io/votable/tests/test_vo.py:181]: Could we just do it for the whole test file?

**Review by mhvk — APPROVED** [review]
Yes, that looks like a better fix than I had!

One question, but nothing depends on it, so I'll approve now.

File: astropy/io/votable/tests/test_vo.py (hunk 1 above)
Comment (neutrinoceros) [reply astropy/io/votable/tests/test_vo.py:181]: In principle yes but there doesn't seem to be a need for it: tests that fail without legacy print options are actually pretty rare in the code base so I don't expect to find more when we remove the global setting, which we should be able to do as soon as we merge #15065

**Comment by neutrinoceros** [comment]
The `io.misc` label should be removed.

**Pull request merged**